core/node/node_framework/src/service/runnables.rs

```diff
@@ -7,9 +7,50 @@ use tokio::sync::Barrier;
 use super::StopReceiver;
 use crate::{
     precondition::Precondition,
-    task::{OneshotTask, Task, UnconstrainedOneshotTask, UnconstrainedTask},
+    task::{OneshotTask, Task, TaskId, UnconstrainedOneshotTask, UnconstrainedTask},
 };
 
+/// Alias for a shutdown hook function type.
+pub trait ShutdownHookFn:
```

Comment:
Dumb question: Since Rust futures do nothing unless polled, wouldn't it suffice to use `BoxFuture<'static, anyhow::Result<()>>` as a shutdown hook? What benefit does enclosing it in a closure provide?

Reply:
It probably would, I just thought that expressing it as a closure provides less cognitive burden, e.g. it's easier to understand that the hook will be invoked later. Storing `Future` until some point in time is not a common operation, so I thought it might be less intuitive.

Reply:
IMO, the interface should be as simple as possible, so a `Future` would work better than a closure returning a future. The fact that the future would be polled eventually can be documented; IMO, they are moderately easy to understand.

Reply:
No objections.